cvat/apps/dataset_manager/annotation.py

```diff
@@ -286,6 +266,74 @@ def to_tracks(self):
         return tracks + shapes.to_tracks()
 
 
+class StreamMerger:
```

Comment:
This class, being effectively a generator, should probably
- support closing
- prohibit addition of new iterables after the iteration is started

Consider returning a new generator from `__iter__()` that remembers the parameters at the moment of calling, it should be simpler to implement than making a generator from scratch.